tensorflow/contrib/distribute/python/moving_averages_test.py

```diff
@@ -136,6 +136,26 @@ def testCrossDevice(self, distribution):
            (2.0 * 0.25 + 0.0) / (1.0 * 0.25 + 1.0)],
           var.eval())
 
+  @combinations.generate(all_combinations)
+  def testAssignVariable(self, distribution):
+    def replica_fn():
+      var = variables.Variable([10.0, 11.0])
+      # Here we expect to check the case when input value are variable.
+      val = variables.Variable([1., 2.])
```

Comment:
Is this case common in your use cases? The update value to moving average is usually different across different replicas. If it is a MirroredVariable, then their components are always in sync and it is not necessary run all-reduce.

Reply:
You are right, here we exactly consider the case when input value is MirroredVariable, while the original codebase cannot accept `Mirrored` type as it will trigger MirroredStrategy._reduce's [assert error](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/distribute/python/mirrored_strategy.py#L618). So this PR kindly fix this bug by adding a convert before calling into `_reduce`.

Reply:
Yuefeng is saying this is not the right fix; it avoids triggering the assert without fixing the underlying issue it was trying to prevent. It is possible we should make "reduce mean" into a no-op for mirrored values, which would fix this issue.

Reply:
@josh11b Thanks for your explanation, it makes sense. The code has been modified accordingly.